application/templates/pages/docs.html

```diff
@@ -7,11 +7,142 @@
+    {% set _schema = params.components['schemas'][_parameterComponentName] %}
+    Must be one of: {{ _schema["enum"][:-1] | join(", ") }} or {{ _schema["enum"] | last }}
+  {%  else %}
+    {{ params.parameter["description"] }}
+  {%  endif %}
+{% endmacro %}
+
+{% macro _renderEndpoint(params={}) %}
+  {% set _parameterRows = [] %}
+
+  <!-- for path in params.paths.keys() -->
+  {% for parameter in params.parameters %}
+    {{
+      _parameterRows.append([
+        {
+          'html': _parameterName({ 'name': parameter.name, 'required': parameter.required }) | safe
+        },
+        {
+          'html': _parameterType({'parameter':parameter,'components':components}) | safe,
+          'classes': 'govuk-!-font-size-14'
+        },
+        {
+          'html': _parameterDescription({'parameter':parameter,'components':components})| safe,
+          'classes': 'govuk-!-font-size-16'
+        }
+      ]) | debug | safe
+    }}
+  {% endfor %}
+
+  {% call appSummaryCard({
+      'titleText': params.name,
+      'classes': 'govuk-!-margin-bottom-6',
+      'attributes': [
+        ['aria-label', params.name]
+      ]
+    }) %}
+
+    <p class="govuk-body">{{ params.description }}</p>
+
+    {{
+      govukTable({
+        'caption': 'Request details',
+        'captionClasses': "govuk-table__caption--s govuk-hint",
+        'attributes': {
+          'aria-label': 'Request details'
+        },
+        'head': [
+          {
+            'text': "Request type",
+            'classes': 'govuk-!-width-one-third',
+            'data': 'data-modules'
+          },
+          {
+            'text': "URL"
+          }
+        ],
+        'rows': [
+          [
+            {
+              'html': govukTag({ 'text': params.requestType, 'classes': ('govuk-tag--red' if (params.requestType | lower) == 'delete' else 'govuk-tag--green') })
+            },
+            {
+              'html': ('<span class="app-code-block">' + params.url + '</span>'),
+              'classes': 'govuk-!-font-size-16'
+            }
+          ]
+        ]
+      })
+    }}
+
+    {% set parametersHTML %}
+      {{
+        govukTable({
+          'caption': "Parameters",
+          'captionClasses': "govuk-table__caption--s govuk-hint",
+          'attributes': {
+            'aria-label': 'Parameters'
+          },
+          'head': [
+            {
+              'text': "Name",
+              'classes': 'govuk-!-font-size-16'
+            },
+            {
+              'text': "Type",
+              'classes': 'govuk-!-font-size-16'
+            },
+            {
+              'text': "Description",
+              'classes': 'govuk-!-width-one-half govuk-!-font-size-16'
+            }
+          ],
+          'rows': _parameterRows
+        })
+      }}
+    {% endset %}
+
+    {{ dlDataTableWrapper({
+      "tableHTML": parametersHTML
+    }) }}
+
+  {% endcall %}
+{% endmacro %}
+
+
```

Comment:
most of this code has just been moved to the top of the file as it makes reading and understanding the html easier